osier/equations.py

```diff
@@ -118,14 +120,18 @@ def objective_from_energy(technology_list, attribute, solved_dispatch_model):
     non_dispatch_techs = get_nondispatchable_techs(technology_list)
     column_names = get_tech_names(technology_list)
     dispatch_results = solved_dispatch_model.results
+    power_units = solved_dispatch_model.power_units
+    time_delta = solved_dispatch_model.time_delta
+    dispatch_values = dispatch_results[column_names].values.T*power_units*time_delta
```

Comment:
What is `dispatch_results[column_names].values.T`? I'm not sure what the `T` is.

Reply:
`.values` converts a dataframe to a numpy array. `.T` gives the transpose of a numpy array.